lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -37,7 +38,26 @@ function getNodeDetailsData() {
 }
 /* c8 ignore stop */
 
-class TraceElements extends Gatherer {
+class TraceElements extends FRGatherer {
+  /** @type {LH.Gatherer.GathererMeta<'Trace'>} */
+  meta = {
+    supportedModes: ['timespan', 'navigation'],
+    dependencies: {Trace: Trace.symbol},
+  }
+
+  /** @type {Map<string, string>} */
+  animationIdToName = new Map();
+
+  constructor() {
+    super();
+    this._onAnimationStarted = this._onAnimationStarted.bind(this);
+  }
+
+  /** @param {LH.Crdp.Animation.AnimationStartedEvent} args */
+  _onAnimationStarted({animation: {id, name}}) {
+    if (name) this.animationIdToName.set(id, name);
+  }
```

Comment:
```suggestion
  /** @param {LH.Crdp.Animation.AnimationStartedEvent} args */
  _onAnimationStarted = ({animation: {id, name}}) => {
    if (name) this.animationIdToName.set(id, name);
  }
```

FWIW, this was my suggestion if you're going to separate already :)

Reply:
Can't do it, need access to `this`

Reply:
that shouldn't be a problem

```bash
$ cat > foo.js <<EOF
class Foo {
  x = 1
  foo = () => this.x
}

console.log(new Foo().foo())
EOF
$ node foo.js
1

Reply:
Unfortunately `eslint` disagrees

Reply:
That's because they don't support class fields https://github.com/eslint/eslint/issues/14343